### ISSUE TITLE
Exclude websocket connections from Average Response Time (µs)

### ISSUE THREAD

**Issue by AnderssonPeter** [containous/traefik#3620]
I'm running some websockets thru Traefik, this causes the Average Response Time (µs) to spike to huge values.

This would be solved if there was some way to exclude websocket connections.

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Træfik!

Issue templates help us help you by providing all necessary information.

Please edit your issue and use the available templates:

- [Feature request](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Feature_request.md)
- [Bug report](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Bug_report.md)

And remember: each time someone ignores the template, a cute little bunny dies.
<details>
<summary>#SaveTheCuteBunny :heart: :rabbit: :heart:</summary>

![southpark - estrella kill rabbit](https://user-images.githubusercontent.com/5674651/27143914-3cea7896-5130-11e7-81f8-02d8ac2da7a7.gif)

![southpark - estrella kill rabbit2](https://user-images.githubusercontent.com/5674651/27143924-49104c72-5130-11e7-82ae-ccedbd5dfa1c.gif)

</details>

**Comment by alexpmorris** [comment]
I'm having this issue as well using `traefik v1.7.0-rc5`.  As you can see, the ms at the end of each log entry is way out of whack.  For normal http requests, it's generally always under a few hundred ms.  

Hopefully this will be addressed soon in an upcoming release.

```
x.x.x.x - - [28/Oct/2018:01:19:45 +0000] "GET /ws HTTP/1.1" 0 0 "-" "Mozilla/5.0" 7 "frontend_ws" "http://127.0.0.1:8100" 1153794ms
x.x.x.x - - [28/Oct/2018:01:24:33 +0000] "GET /ws HTTP/1.1" 0 0 "-" "Mozilla/5.0" 44 "frontend_ws" "http://127.0.0.1:8100" 1083769ms
x.x.x.x - - [28/Oct/2018:01:19:44 +0000] "GET /ws HTTP/1.1" 0 0 "-" "Mozilla/5.0" 4 "frontend_ws" "http://127.0.0.1:8100" 1938724ms
x.x.x.x - - [28/Oct/2018:01:52:05 +0000] "GET /ws HTTP/1.1" 0 0 "-" "Mozilla/5.0" 57 "frontend_ws" "http://127.0.0.1:8100" 105540ms
```

And this is what my dashboard looks like each time I initiate a new websocket connection:

![image](https://user-images.githubusercontent.com/5326789/47611305-e6ccb800-da38-11e8-8264-33b5bcb1c821.png)

**Comment by siyu6974** [comment]
@mmatur I've pinned down the issue to be related to the thoas/stats package and I'm able to fix it. Unit test goes fine and I'm more than willing to do a PR but the question is should I make the changes to this project directly (under `vendor/`) or should I PR to the original project?

**Comment by ldez** [comment]
@siyu6974 open a PR to the original project.
Link your PR with this issue.

**Comment by siyu6974** [comment]
@ldez  PR opened!
[link here](https://github.com/thoas/stats/pull/29)

**Comment by siyu6974** [comment]
@ldez PR https://github.com/thoas/stats/pull/29 merged.

**Comment by siyu6974** [comment]
I'm unfortunately unable to use `dep` to update the vendor... it kept saying 

```
Solving failure: No versions of github.com/containous/traefik-extra-service-fabric met constraints:
	v1.3.0: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.3.0, as it has a dependency on github.com/jjcollinge/servicefabric with constraint 8eebe170fa1ba25d3dfb928b3f86a7313b13b9fe, which has no overlap with existing constraint master from (root)
	v1.2.0: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.2.0, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.5: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.5, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.4: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.4, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.3: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.3, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.2: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.2, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.1: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.1, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1.0: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1.0, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.6: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.6, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.5: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.5, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.4: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.4, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.3: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.3, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.2: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.2, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.1: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.1, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0.0: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0.0, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	master: Could not introduce github.com/containous/traefik-extra-service-fabric@master, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	init-provider: Could not introduce github.com/containous/traefik-extra-service-fabric@init-provider, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	integration: Could not introduce github.com/containous/traefik-extra-service-fabric@integration, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	integrationv2: Could not introduce github.com/containous/traefik-extra-service-fabric@integrationv2, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.0: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.0, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.
	v1.1: Could not introduce github.com/containous/traefik-extra-service-fabric@v1.1, as it is not allowed by constraint ^1.3.0 from project github.com/containous/traefik.

```

**Comment by ldez** [comment]
@siyu6974 use the branch v1.7 :wink:

**Comment by traefiker** [comment]
Closed by #4313.

**Comment by timoschwarzer** [comment]
It seems like this issue still applies to the Prometheus metrics. Is that intended behavior?
